homeassistant/components/light/hue.py

```diff
@@ -174,7 +178,11 @@ def process_lights(hass, api, bridge, bridge_type, update_lights_cb):
             new_lights.append(bridge.lights[light_id])
         else:
             bridge.lights[light_id].info = info
-            bridge.lights[light_id].schedule_update_ha_state()
+            hass.helpers.dispatcher.dispatcher_send(
+                hass,
```

Comment:
@pvizeli AFAICT this can't work reliably.
I'm supposed to remove this argument when calling via helpers (right), since there's a `@bind_hass` annotation which will take care of this implicitly.

But when we can `dispatcher_send` the entity hasn't necessarily been added yet (or there's a race condition anyway, per @MartinHjelmare's comment on #11183. Because of that `hass.helpers.dispatcher.dispatcher_send` hasn't been bound to its `hass` variable, so this will definitely fail.

Reply:
Remove the `hass` argument as you also write. The dispatcher connect function will protect from a race.

Reply:
Hmm interesting, per my (barely intelligible) comment, I tried that last night and was getting errors. It's working now so I was clearly just too tired :D